test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -10110,5 +10110,27 @@ TEST_F(AtenXlaTensorTest, TestAmpUpdateScale) {
                        cpp_test::GetIgnoredCounters());
 }
 
+TEST_F(AtenXlaTensorTest, TestEarlySyncLiveTensors) {
+  torch::Tensor scalar_tensor =
+      torch::scalar_tensor(1., torch::TensorOptions(torch::kFloat));
+  torch::Scalar scalar1 = scalar_tensor.item();
+  ForEachDevice([&](const torch::Device& device) {
+    torch::Tensor xla_scalar_tensor = CopyToDevice(scalar_tensor, device);
+    torch::Scalar scalar2 = xla_scalar_tensor.item();
+    ASSERT_EQ(scalar1.to<float>(), scalar2.to<float>());
+  });
+  static bool sync =
+      xla::sys_util::GetEnvBool("XLA_SYNC_BEFORE_ITEM_CALL", true);
```

Comment:
Any reason you set the default value to true here?  I think this test can serve the purpose that when we tell user to try to feautre, they can use this test to verify if `earlySync` works.

Reply:
I think if we set this value to False, then this test is essentially not testing anything. If we set to True, it tests if early sync happened. WDYT? Should I set to False?

Reply:
The default value only matters if `XLA_SYNC_BEFORE_ITEM_CALL` is not being set. 

You have 
```
export XLA_SYNC_BEFORE_ITEM_CALL=0
```
in `test/cpp/run_tests.sh` so the default value here does not matter when running all cpp test through `run_tests.sh`.

However if user did not set `XLA_SYNC_BEFORE_ITEM_CALL` and manually run `TestEarlySyncLiveTensors`. Test will expect `EarlySyncLiveTensorsCount` counter to increase while `AtenXlaType::_local_scalar_dense` do nothing since it has
```
  static bool sync =
      xla::sys_util::GetEnvBool("XLA_SYNC_BEFORE_ITEM_CALL", false);
```

The default value of a `get_env` has to match in this case.

Reply:
If you want to test the `early_sync` through this test, you will want to manually set the env var before calling `item` and unset it before exiting the test.

Reply:
what do we do for experimental dynamic shape tests? I guess it would be ok to replicate that behavior here.

Reply:
For cpp test, we always run with `dynamic_shpae` enabled. For python test, we enable `dynamic_shape` for a subset of tests. 

I don't think we should enable `earyly_sync` for all op tests since some of them is view related. I think one thing we could do is to run you test separatlly with `early_sync` flag on.

Adding a line under 
https://github.com/pytorch/xla/blob/9c2f91e3c205e2735c4fbcc4348cee6a74e029ff/test/cpp/run_tests.sh#L61
and https://github.com/pytorch/xla/blob/9c2f91e3c205e2735c4fbcc4348cee6a74e029ff/test/cpp/run_tests.sh#L63

and run 
```
XLA_SYNC_BEFORE_ITEM_CALL=1 ./test_ptxla --gtest_filter=AtenXlaTensorTest.TestEarlySyncLiveTensors
```
should work I think.

Reply:
Could you update the default value to be false?

Reply:
done. thanks.